### ISSUE TITLE
Can we keep the smithy-rs docker image on the repo with Git LFS?

### ISSUE THREAD

**Issue by thomas-k-cameron** [awslabs/smithy-rs#2266]
We are getting a lot of errors from the ecr lately.
```
Error response from daemon: toomanyrequests: Data limit exceeded
```

https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage
Can we use git LFS to keep the docker image within this repository?

Assuming that image size is around 1gb, compressing it with zstd(level 15, or something like it) should turn it into 20 to 30mb or something.   

I think keeping the whole thing within the repository without LFS isn't totally out of the question.

@jdisanti 
I'm going to mention you because I saw you creating a PR to address this issue.

**Comment by jdisanti** [comment]
That's an interesting idea. @rcoh is trying out authenticating the requests to ECR to reduce the throttling in https://github.com/awslabs/smithy-rs/pull/2265. If that doesn't improve things, we will explore more options.

**Comment by thomas-k-cameron** [comment]
I closed it since #2265 seemed to have worked.
